charmhub/client.go

```diff
@@ -39,6 +39,8 @@ const (
 	CharmHubServerVersion = "v2"
 	CharmHubServerEntity  = "charms"
 
+	MetadataHeader = "Juju-Metadata"
```

Comment:
We should originally have made this X-Juju-Metadata
Ideally we'd use the correct one for CH moving forward. Something to consider.

Reply:
Updated. Per discussion, I'll also ask Simon and Heather what the charmhub team wants.